src/lambdaconnect_model/scoping.clj

```diff
@@ -194,98 +199,170 @@
                   [#{} (none-rules '?user symbol)])
 
                 (let [op (first rule)]
-                  (if (list? rule)
-                    (let [internal-wheres (map #(where-for-rule % false #{}) (rest rule))
-                          dependency-list (map nested-dependencies (map first internal-wheres))
-                          common-dependencies (reduce (if (= op 'and) union intersection) dependency-list)
-                          final-wheres (map #(where-for-rule % false common-dependencies) (rest rule))
-                          internal-queries (reduce concat (map second final-wheres))
-                          particular-dependencies (map #(difference % common-dependencies ignored-dependencies) dependency-list)
-                          zipped (map vector (map second final-wheres) particular-dependencies)
-                          dependent-entities-from-common (union #{'?user entity-symbol} (set (map symbol-for-tag common-dependencies)))
-                          appended (reduce concat (map (fn [[where dep]]
-                                                         (let [rules (reorder-rules (set (dependent-rules dep common-dependencies where)) #{entity-symbol})]
-                                                           (if (and (> (count rules) 1) (= op 'or)) [`(~'and ~@rules)] rules))) zipped))
-                          used-entities (reduce union (map (fn [[where dep]] (find-all-symbols (dependent-rules dep common-dependencies where))) zipped))
-                          dependent-entities (intersection dependent-entities-from-common used-entities)
-                          ;; _ (println "OP: " op "rule: " rule)
-                          ;; _ (println "DEPENDENT: " dependent-entities-from-common)
-                          ;; _ (println "USED: " used-entities)
-                          ;; _ (println "INTerSeCTION: " (intersection dependent-entities-from-common used-entities))
+                  (if (= 'allowed-if op)
+                    (let [const-key (-> 
+                                     rule
+                                     second
+                                     name
+                                     keyword)
+                          _ (assert (contains? constants const-key) (str "The rule " rule " contains a constant " const-key " not present in constants map."))
+                          const (const-key constants)
+                          const-val (if (delay? const) @const const) 
```

Comment:
Lepiej `(instance? clojure.lang.IDeref const)`, wtedy też możesz podać future, atom, ...

Reply:
Ok, jest też funkcja https://clojuredocs.org/clojure.core/force, możesz zamienić tego ifa na `force`, jeśli uważasz to za czytelniejsze